js/ui/switch.js

```diff
@@ -162,34 +171,35 @@ var Switch = Editor.inherit({
         this._renderSwipeable();
 
         this.callBase();
-    },
 
-    _render: function() {
         this._renderSwitchInner();
         this._renderLabels();
-
-        this._renderHandleWidth();
         this._renderValue();
-
-        this.callBase();
-    },
-
-    _renderHandleWidth: function() {
-        this._handleWidth = parseFloat(window.getComputedStyle(this._$handle.get(0)).width);
     },
 
     _getCalcOffset: function(value, offset) {
-        var ratio = offset - Number(!value);
-        return "calc(" + 100 * ratio + "% + " + -this._getHandleWidth() * ratio + "px)";
+        var ratio = (offset - this._offsetDirection() * Number(!value)) / 2;
+        return 100 * ratio + "%";
     },
 
-    _getHandleWidth: function() {
-        !this._handleWidth && this._renderHandleWidth();
-        return this._handleWidth;
-    },
+    _getHandleOffset: function(value, offset) {
+        var calcValue;
 
-    _getPixelOffset: function(value, offset) {
-        return this._getMarginBound() * (offset - Number(!value));
+        if(this.option("rtlEnabled")) {
+            if(!value) {
+                calcValue = -100 + 100 * (-offset);
+                return calcValue + "%";
+            } else {
+                return 100 * (-offset) + "%";
+            }
+        } else {
+            if(value) {
+                calcValue = -100 + 100 * (-offset);
+                return calcValue + "%";
+            } else {
+                return 100 * (-offset) + "%";
```

Comment:
Maybe extract this duplicate code to the method,
I am talking about:
`
if(value) {
     calcValue = -100 + 100 * (-offset);
      return calcValue + "%";
} else {
       return 100 * (-offset) + "%";
` 

An input value parameter should be according to the rtlEnabled.